src/ImageSharp/Common/Memory/BufferPointer{T}.cs

```diff
@@ -74,6 +74,20 @@ public BufferPointer(T[] array, void* pointerToArray)
         public IntPtr PointerAtOffset { get; private set; }
 
         /// <summary>
+        /// Gets the element at the specified position.
+        /// </summary>
+        /// <param name="index">The index from the start of this Pointer to the required element.</param>
+        /// <returns>The <see typeparam="T"/> at the specified position.</returns>
+        public T this[int index]
```

Comment:
It should! Can be done with `Unsafe.AsRef<T>(ptr)`. 
This does not effect C# < 7 users here, because BufferPointer is internal.

Happy you spotted this change because, I'm turning `BufferPointer <T>` into `BufferSpan<T>` at the moment. Also wanted to add this indexer, but let's merge this first then to have less conflicts :)

BTW: `Unsafe.AsRef<T>(ptr)` is not visible for C#6 users. It means that `Span<T>` will be a ... VS2017-only data type!!! I have no idea how does Microsoft plan to manage this....